### PR TITLE
Migrate to kubebuilder

### PR DESCRIPTION
**Pull request by orbatschow**
This PR will introduce some breaking changes:

- Migration to kubebuilder
- CI/CD rework
- lift the CRDs to v1beta1

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  quality:
+    runs-on: ubuntu-latest
+    steps:
+      - name: compute checksums of auto generated files
+        run: |
+          echo hello world
+
+      - name: lint
+        run: | 
+          make test
```
Comment (ntap-fge) [inline .github/workflows/main.yaml:20]: lint instead of test?

Comment (orbatschow) [reply .github/workflows/main.yaml:20]: yes, i temporarily set this to test, as some linting errors are still alive

**Comment by orbatschow** [comment]
Fixes https://github.com/orbatschow/kubepost/issues/42

File: .github/workflows/main.yaml (hunk 1 above)
Comment (orbatschow) [reply .github/workflows/main.yaml]: Fixed

**Pull request merged**